platform/core/src/studies/services/wado/seriesLoadQueue.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Module: seriesLoadQueue.js
+ * This module represents a queue of series from a given study to be loaded.
+ * It also implements the pub/sub interfaece.
+ */
+
+import { api } from 'dicomweb-client';
+import { makePubSub } from '../../../lib/pubSub';
+
+const STUDY_ID = Symbol('StudyId');
+const SERIES_IDS = Symbol('SeriesIds');
+const DICOM_WEB_CLIENT = Symbol('DicomWebClient');
+const EVENT_SERIES_LOADED = 'seriesLoaded';
+const EVENT_SERIES_LOAD_ERROR = 'seriesLoadError';
+
+const seriesLoadQueueProto = makePubSub({
+  EVENT_SERIES_LOADED,
+  EVENT_SERIES_LOAD_ERROR,
+  enqueue(entry) {
+    return enqueue(this, entry);
+  },
+  dequeue() {
+    return dequeue(this);
+  },
+  size() {
+    return size(this);
+  },
+});
+
+function enqueue(queue, seriesId) {
+  if (isValidId(seriesId)) {
+    queue[SERIES_IDS].unshift(seriesId);
+    return true;
+  }
+  return false;
+}
+
+function dequeue(queue) {
```

Comment:
The naming here is interesting.
I don't know that I would expect `dequeue` to `retrieveSeriesMetadata`
It feels like a `queue` implementation that has a side-effect
That then also publishes an event that causes a different side-effect.
Very hard to follow

Reply:
As the name implies this is not a regular/simple queue but a `series load queue` instead which from my perspective implies that dequeueing its items would effectively load the queued series...
But no strong opinion on how I should name it. Totally open to suggestions.